### PR TITLE
[js/webgpu] Increase workgroupSize if only one workgroup is dispached

### PR DESCRIPTION
**Pull request by qjia7**
#22031

For reduce related ops, we should increase workgroupSize to improve parallelism if only one workgroup is dispatched.

The total ReduceMean time becomes 8.98 ms from 77.79 ms on my iGPUs.

### CODE REVIEW

**Comment by qjia7** [comment]
@guschmue @fs-eire Please take a look, thanks.

File: js/web/lib/wasm/jsep/webgpu/ops/reduce-shared.ts [hunk 1]
```diff
@@ -188,7 +192,7 @@ export const createReduceSharedProgramInfo = (
   // One work group is responsible for only one element of output.
   return {
     name,
-    shaderCache,
+    shaderCache: { hint: `${cacheKey};${workgroupSize}`, inputDependencies: ['type'] },
```
Comment (fs-eire) [inline js/web/lib/wasm/jsep/webgpu/ops/reduce-shared.ts:195]: i think this is one place that JSEP is different from WebGPU EP.

in JSEP, WG size is not included in cache by default, but WebGPU EP it is.

Suggest to add a comments here for reference.

**Review by guschmue — DISMISSED** [review]

File: js/web/lib/wasm/jsep/webgpu/ops/reduce-shared.ts (hunk 1 above)
Comment (qjia7) [reply js/web/lib/wasm/jsep/webgpu/ops/reduce-shared.ts:196]: Done

**Review by guschmue — APPROVED** [review]

**Pull request merged**